rolling/arithmetic/nunique_indexed.py

```diff
@@ -0,0 +1,72 @@
+from collections import Counter, deque
+
+from rolling.base_indexed import RollingIndexed
+
+class NuniqueIndexed(RollingIndexed):
+    """
+    Iterator object that counts the number of unique values in a rolling
+    window with an index array
+
+    Parameters
+    ----------
+    
+    index : the object that will serve as index
+    iterable : any iterable object
+    window_size : same type as the index, the maximum size (difference between indices)
+        of the rolling window moving over the iterable
+    
+    Complexity
+    ----------
+    
+    Update time: O(1)
+    Memory usage: O(k)
+    
+    where k is the size of the rolling window (which can potentially be n)
+
+    """
+
+    def __init__(self, index, iterable, window_size, window_type="variable"):
+        self._idx_buffer = deque()
+        self._val_buffer = deque()
+        self._counter = Counter()
+        self._nunique = 0
+        super().__init__(index, iterable, window_size, window_type)
+
+    def _init_variable(self, **kwargs):
+        pass
+
+    def _insert(self, idx, val):
+        assert not self._idx_buffer or self._idx_buffer[0] <= idx, \
+            "Indices should be monotonic"
```

Comment:
```suggestion
        if self._idx_buffer and self._idx_buffer[0] > idx:
            raise ValueError("Indices should be monotonic")
```
I guess this would be an error with the user's data rather than error in the internal logic of the code, so a raising a `ValueError` seems like a better fit than using `assert`.